pkg/iam/iam.go

```diff
@@ -67,47 +68,138 @@ func New(config IAMServiceConfig) (*IAMService, error) {
 	return s, nil
 }
 
-func (s *IAMService) Reconcile() error {
+func (s *IAMService) ReconcileRole() error {
 	s.log.Info("reconciling IAM role")
 
-	err := s.createMainRole()
+	params := struct {
+		ClusterName      string
+		EC2ServiceDomain string
+	}{
+		ClusterName:      s.clusterName,
+		EC2ServiceDomain: ec2ServiceDomain(s.region),
+	}
+
+	err := s.reconcileRole(s.mainRoleName, params)
+	if err != nil {
+		return err
+	}
+
+	s.log.Info("finished reconciling IAM role")
+	return nil
+}
+
+func (s *IAMService) ReconcileKiamRole() error {
+	s.log.Info("reconciling KIAM IAM role")
+
+	var controlPlaneRoleARN string
+	{
+
+		i := &awsiam.GetRoleInput{
+			RoleName: aws.String(s.mainRoleName),
+		}
+
+		o, err := s.iamClient.GetRole(i)
+		if err != nil {
+			s.log.Error(err, "failed to fetch get ControlPlane role")
+			return err
+		}
+
+		controlPlaneRoleARN = *o.Role.Arn
+	}
+
+	params := struct {
+		ControlPlaneRoleARN string
+		EC2ServiceDomain    string
+	}{
+		ControlPlaneRoleARN: controlPlaneRoleARN,
+		EC2ServiceDomain:    ec2ServiceDomain(s.region),
+	}
+
+	err := s.reconcileRole(roleName(KIAMRole, s.clusterName), params)
+	if err != nil {
+		return err
+	}
+
+	s.log.Info("finished reconciling IAM role")
+	return nil
+}
+
+func (s *IAMService) ReconcileRoute53Role() error {
+	s.log.Info("reconciling KIAM IAM role")
+
+	var kiamRoleARN string
+	{
+		i := &awsiam.GetRoleInput{
+			RoleName: aws.String(roleName(KIAMRole, s.clusterName)),
+		}
+
+		o, err := s.iamClient.GetRole(i)
+		if err != nil {
+			s.log.Error(err, "failed to fetch get KIAM role")
```

Comment:
```suggestion
			s.log.Error(err, "failed to fetch KIAM role")
```

Reply:
Is there a possibility that kiam is diasbled and route 53 is enabled? Because in that case we would fail here, right?

Reply:
yeah, if KIAM role is disabled then it  would most probably fail here, maybe we should enable route53 only if both are nebaled

Reply:
Yeah, I think that would be more safe. :)

Reply:
done